### PR TITLE
Fun facts are fetched from the server

### PR DESCRIPTION
**Pull request by thefredo1000**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] PR has a descriptive title
- [ ] PR contains closely related changes only (should not have many unrelated changes)
- [ ] STEP pod members have been added as reviewers
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: ayup/portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -38,3 +38,14 @@ function addRandomFact() {
   factContainer.innerHTML = newFact;
 }
 
+/**
+ * Fetch the facts from the server
+ */
+async function getRandomFactUsingAsyncAwait() {
+  const response = await fetch('/data');
+  const fact = await response.text();
+  console.log()
```
Comment (seanjohnite) [inline ayup/portfolio/src/main/webapp/script.js:47]: remove?

File: ayup/portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -38,3 +38,14 @@ function addRandomFact() {
   factContainer.innerHTML = newFact;
 }
 
+/**
+ * Fetch the facts from the server
+ */
+async function getRandomFactUsingAsyncAwait() {
+  const response = await fetch('/data');
+  const fact = await response.text();
```
Comment (seanjohnite) [inline ayup/portfolio/src/main/webapp/script.js:46]: I don't think you need this second await. Do you?

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,9 +26,25 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> facts;
+
+  @Override
+  public void init() {
+    facts = new ArrayList<>();
+    facts.add("I develop Nintendo GameBoy Advance ROMs in my spare time");
+    facts.add("I was a YouTube star back in the days <a href='https://youtube.com/thefredo1000'>Check out my channel</a>");
```
Comment (seanjohnite) [inline ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: You need to do something special to the text if you want this to work.. Not exactly sure what it is, but I ran your server and it doesn't actually link, just sets the text: 
image

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -24,9 +26,25 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> facts;
+
+  @Override
+  public void init() {
```
Comment (seanjohnite) [inline ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: You might be able to get away with this fact list being a static private ArrayList?

File: ayup/portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (brettapeters) [reply ayup/portfolio/src/main/webapp/script.js:46]: `response.text()` returns a promise.

https://developer.mozilla.org/en-US/docs/Web/API/Body/text

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (brettapeters) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: +1. I don't think this is the intended use of the `init()` method. https://www.gnu.org/software/classpathx/servletapi/javadoc/javax/servlet/GenericServlet.html

File: ayup/portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (thefredo1000) [reply ayup/portfolio/src/main/webapp/script.js:47]: Done

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (thefredo1000) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: Thanks for noting that, I didn´t notice.

File: ayup/portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (brettapeters) [reply ayup/portfolio/src/main/webapp/script.js:47]: Push new commit?

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (thefredo1000) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: So where do I add the facts to the fact list?

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (thefredo1000) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: Done

File: ayup/portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (thefredo1000) [reply ayup/portfolio/src/main/webapp/script.js]: My bad, forgot to push... 😑

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (brettapeters) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: Usually like this.
```java
private static final ImmutableList<String> FUN_FACTS =
  ImmutableList.of(
      "fact 1", "fact 2", ... );
```

Comment (thefredo1000) [reply ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: I see, thanks!

File: ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -24,9 +26,19 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final ImmutableList<String> FUN_FACTS = ImmutableList.of(
```
Comment (brettapeters) [inline ayup/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: I guess if you didn't want to bring in the Guava dependency you'd do something like...

```java
private static final ArrayList<String> FUN_FACTS =
      new ArrayList<>(Arrays.asList("fact 1", "fact 2", ...));
```

Guava is used in basically every Java project at Google.

**Pull request merged**